src/Compilers/CSharp/Test/Emit2/CodeGen/CodeGenNullCheckedParameterTests.cs

```diff
@@ -1738,5 +1776,217 @@ .maxstack  1
+    public override bool Equals(object obj) => false;
+    public override int GetHashCode() => 0;
+
+    public static void M(C c!!)
+    {
+        if (c == null) { System.Console.Write(1); }
+    }
+}
+";
+            var verifier = CompileAndVerify(source);
+            verifier.VerifyDiagnostics();
+            verifier.VerifyIL("C.M", @"
+{
+  // Code size       27 (0x1b)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldstr      ""c""
+  IL_0006:  call       ""ThrowIfNull""
+  IL_000b:  ldarg.0
+  IL_000c:  ldnull
+  IL_000d:  call       ""bool C.op_Equality(C, C)""
+  IL_0012:  brfalse.s  IL_001a
+  IL_0014:  ldc.i4.1
+  IL_0015:  call       ""void System.Console.Write(int)""
+  IL_001a:  ret
+}");
+        }
+
+        [Fact]
+        public void Override_NullCheckedDifference()
+        {
+            var source =
+@"
+class Base
+{
+    public virtual void M1(string x!!) { }
+    public virtual void M2(string x) { }
+}
+
+class Derived : Base
+{
+    public override void M1(string x) { }
+    public override void M2(string x!!) { }
+}";
+            var verifier = CompileAndVerify(source);
+            verifier.VerifyDiagnostics();
+
+            verifier.VerifyIL("Base.M1", @"
+{
+  // Code size       12 (0xc)
+  .maxstack  2
+  IL_0000:  ldarg.1
+  IL_0001:  ldstr      ""x""
+  IL_0006:  call       ""ThrowIfNull""
+  IL_000b:  ret
+}");
+            verifier.VerifyIL("Base.M2", @"
+{
+  // Code size        1 (0x1)
+  .maxstack  0
+  IL_0000:  ret
+}");
+            verifier.VerifyIL("Derived.M1", @"
+{
+  // Code size        1 (0x1)
+  .maxstack  0
+  IL_0000:  ret
+}");
+            verifier.VerifyIL("Derived.M2", @"
+{
+  // Code size       12 (0xc)
+  .maxstack  2
+  IL_0000:  ldarg.1
+  IL_0001:  ldstr      ""x""
+  IL_0006:  call       ""ThrowIfNull""
+  IL_000b:  ret
+}");
+        }
+
+        [Fact]
+        public void Implementation_NullCheckedDifference()
+        {
+            var source =
+@"
+interface Base
+{
+    public void M1(string x!!) { }
+    public void M2(string x) { }
+}
+
+class Derived : Base
+{
+    public void M1(string x) { }
+    public void M2(string x!!) { }
+}";
+            var verifier = CompileAndVerify(source, targetFramework: TargetFramework.NetCoreApp);
+            verifier.VerifyDiagnostics();
+
+            verifier.VerifyIL("Base.M1", @"
+{
+  // Code size       12 (0xc)
+  .maxstack  2
+  IL_0000:  ldarg.1
+  IL_0001:  ldstr      ""x""
+  IL_0006:  call       ""ThrowIfNull""
+  IL_000b:  ret
+}");
+            verifier.VerifyIL("Base.M2", @"
+{
+  // Code size        1 (0x1)
+  .maxstack  0
+  IL_0000:  ret
+}");
+            verifier.VerifyIL("Derived.M1", @"
+{
+  // Code size        1 (0x1)
+  .maxstack  0
+  IL_0000:  ret
+}");
+            verifier.VerifyIL("Derived.M2", @"
+{
+  // Code size       12 (0xc)
+  .maxstack  2
+  IL_0000:  ldarg.1
+  IL_0001:  ldstr      ""x""
+  IL_0006:  call       ""ThrowIfNull""
+  IL_000b:  ret
+}");
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider checking implicit and explicit implementation of an interface method with no implementation.

Reply:
I think I did this.